moveit_core/trajectory_processing/test/test_time_optimal_trajectory_generation.cpp

```diff
@@ -227,6 +229,98 @@ TEST(time_optimal_trajectory_generation, testLargeAccel)
+  moveit::core::RobotState waypoint_state(robot_model);
+  waypoint_state.setToDefaultValues();
+
+  robot_trajectory::RobotTrajectory trajectory(robot_model, group);
+  waypoint_state.setJointGroupPositions(group, { -0.5, -3.52, 1.35, -2.51, -0.88, 0.63, 0.0 });
+  trajectory.addSuffixWayPoint(waypoint_state, 0.1);
+  waypoint_state.setJointGroupPositions(group, { -0.5, -3.52, 1.35, -2.51, -0.88, 0.63, 0.0 });
+  trajectory.addSuffixWayPoint(waypoint_state, 0.1);
+  waypoint_state.setJointGroupPositions(group, { 0.0, -3.5, 1.4, -1.2, -1.0, -0.2, 0.0 });
+  trajectory.addSuffixWayPoint(waypoint_state, 0.1);
+  waypoint_state.setJointGroupPositions(group, { -0.5, -3.52, 1.35, -2.51, -0.88, 0.63, 0.0 });
+  trajectory.addSuffixWayPoint(waypoint_state, 0.1);
+  waypoint_state.setJointGroupPositions(group, { 0.0, -3.5, 1.4, -1.2, -1.0, -0.2, 0.0 });
+  trajectory.addSuffixWayPoint(waypoint_state, 0.1);
+  waypoint_state.setJointGroupPositions(group, { -0.5, -3.52, 1.35, -2.51, -0.88, 0.63, 0.0 });
+  trajectory.addSuffixWayPoint(waypoint_state, 0.1);
+
+  // Number TOTG iterations
+  constexpr size_t totg_iterations = 10;
+
+  // Test computing the dynamics repeatedly with the same totg instance
+  moveit_msgs::RobotTrajectory first_trajectory_msg_start, first_trajectory_msg_end;
+  {
+    robot_trajectory::RobotTrajectory test_trajectory(trajectory, true /* deep copy */);
+
+    TimeOptimalTrajectoryGeneration totg;
+    ASSERT_TRUE(totg.computeTimeStamps(test_trajectory)) << "Failed to compute time stamps";
+
+    test_trajectory.getRobotTrajectoryMsg(first_trajectory_msg_start);
+
+    // Iteratively recompute timestamps with same totg instance
+    for (size_t i = 0; i < totg_iterations; ++i)
+    {
+      bool totg_success = totg.computeTimeStamps(test_trajectory);
+      EXPECT_TRUE(totg_success) << "Failed to compute time stamps with a same TOTG instance in iteration " << i;
+
+      if (totg_success)
+        break;
+    }
+
+    test_trajectory.getRobotTrajectoryMsg(first_trajectory_msg_end);
+  }
+
+  // Test computing the dynamics repeatedly with one TOTG instance per call
+  moveit_msgs::RobotTrajectory second_trajectory_msg_start, second_trajectory_msg_end;
+  {
+    robot_trajectory::RobotTrajectory test_trajectory(trajectory, true /* deep copy */);
+
+    {
+      TimeOptimalTrajectoryGeneration totg;
+      ASSERT_TRUE(totg.computeTimeStamps(test_trajectory)) << "Failed to compute time stamps";
+    }
+
+    test_trajectory.getRobotTrajectoryMsg(second_trajectory_msg_start);
+
+    // Iteratively recompute timestamps with new totg instances
+    for (size_t i = 0; i < totg_iterations; ++i)
+    {
+      TimeOptimalTrajectoryGeneration totg;
+      bool totg_success = totg.computeTimeStamps(test_trajectory);
+      EXPECT_TRUE(totg_success) << "Failed to compute time stamps with a new TOTG instance in iteration " << i;
+
+      if (totg_success)
+        break;
```

Comment:
@henningkayser you force-pushed over a commit of mine that fixes this exit condition. Probably none of the loops should have the exit condition

Reply:
Thanks, fixed